### PR TITLE
classifier freeze and figure updates

### PR DESCRIPTION
**Pull request by gwaybio**
Minor changes to classifier hyperparameters and figures. Added protein quantification file for validation set and cleaned up code.

### CODE REVIEW

File: scripts/viz/viz_validation.r [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # 2) Scatterplot of predictions by quantified protein
 # 3) Boxplot of predictions for NF1 WT/Inactive protein
 
-set.seed(123)
+set.seed(1234)
```
Comment (dhimmel) [inline scripts/viz/viz_validation.r]: Is this seed just for geom_jitter? If so, I'd place it closer to that function and add a comment about what this is for.

File: scripts/viz/viz_validation.r [hunk 2]
```diff
@@ -133,22 +135,24 @@ predict_nf1[plot_ready$weight_auc <= 0] <- 'NF1 Wildtype'
...
+predict_plot$predict_nf1 <- factor(predict_plot$predict_nf1, 
+                                   levels = c('NF1 Wildtype', 'NF1 Inactive'))
 
 # Reassign CB2 and H5M
 predict_plot <- predict_plot[1:(nrow(predict_plot) - 2), ]
 predict_plot[predict_plot$sample_id == 'H5M', 'u87pi.norm'] <- h5m_mean
 predict_plot[predict_plot$sample_id == 'CB2', 'u87pi.norm'] <- cb2_mean
 
-png(figure_pred, height = 370, width = 450)
+png(figure_pred, height = 250, width = 360)
```
Comment (dhimmel) [inline scripts/viz/viz_validation.r:147]: Consider `ggsave` for future.

File: data/protein_quantification.csv [hunk 3]
```diff
@@ -0,0 +1,27 @@
+,u87pi norm,h5m norm,plate
```
Comment (dhimmel) [inline data/protein_quantification.csv]: Having a blank name for this first column is not good. Not sure if you create this file or not

File: scripts/viz/viz_validation.r (hunk 1 above)
Comment (gwaybio) [inline scripts/viz/viz_validation.r]: yep! will do

File: data/protein_quantification.csv (hunk 3 above)
Comment (gwaybio) [inline data/protein_quantification.csv]: I don't create it - collaborators at Dartmouth have (i use it in an R script for plotting only) `scripts/viz/viz_validation.R`

I can add a column name here: `sample_id`

**Comment by dhimmel** [comment]
> Minor changes to classifier hyperparameters

Are you not using cross validation to choose optimal hyperparameters?

File: data/protein_quantification.csv (hunk 3 above)
Comment (dhimmel) [inline data/protein_quantification.csv]: Yeah I'm in favor of adding the `sample_id` column as long as the manual edit is tracked by the git history (which it will be).

**Comment by gwaybio** [comment]
of course I am! :smile:

**Comment by dhimmel** [comment]
> of course I am! :smile:

What are these hardcoded parameters?

```
optimal_alpha=0.15
optimal_l1=0.1
```

**Comment by gwaybio** [comment]
A result of parameter sweep - the data splits are super variable and for some reason setting a seed does not overcome difference when cross validation splits training/testing.

nf1_supplementary_figure_s1

**Comment by dhimmel** [comment]
AUROC is independent of positive prevalence... so balanced qualifier isn't needed? Are you saying given the variability you need the human interference?

If your selection of optimal requires a manual oversight then hard coding makes sense. Otherwise you shouldn't need to hard code.

**Comment by gwaybio** [comment]
> so balanced qualifier isn't needed?

Good point - I should remove `balanced`

> Are you saying given the variability you need the human interference?

The hyperparameter space of optimal performance is fairly large - and in separate research it seems that this problem cannot be overcome with standard feature construction or alternative algorithms. (I am fairly confident that the poor performance is a direct result of incorrect labeling!) Because the hyperparameter space is large, small variations in splits results in different "optimal" l1 and alpha params. I tried several combinations - but in order to keep a consistent classifier for our collaborators (who are analyzing the important gene contributions) I felt it was best to ensure consistency in the models.

**Comment by allaway** [comment]
Let me know if you're anticipating any changes in the gene lists you sent over yesterday - if so I will hold off on rewriting results!

**Comment by gwaybio** [comment]
@allaway I am not anticipating any changes at the moment - but lets hold off for now, just to be safe. I think we'll be all set once the pull request is merged. Then we can set a finalized frozen gene list.

**Comment by dhimmel** [comment]
If you want to keep a tidy commit history, it would have been ideal to structure 60fd13e99527aaf73f6af017d4cb9b142ea70f86 as three different commits. Not a big deal but for future reference, commits that do many different things are less fashionable these days.

**Comment by gwaybio** [comment]
good to know! thanks :octocat:

**Comment by dhimmel** [comment]
> I am fairly confident that the poor performance is a direct result of incorrect labeling!

Sample mixup or measurement error in the outcome?

> Because the hyperparameter space is large, small variations in splits results in different "optimal" l1 and alpha params.

For future reference, I'd consider **repeated** cross-validation in this situation.

> to keep a consistent classifier for our collaborators (who are analyzing the important gene contributions) I felt it was best to ensure consistency in the models.

You're life will be easier if you only have to choose the regularization strength and not the mixing parameter. Consider [`LogisticRegressionCV`](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegressionCV.html) (but makes you pick between lasso and ridge).

**Comment by dhimmel** [comment]
Pull request looks good to me.

**Comment by gwaybio** [comment]
> Sample mixup or measurement error in the outcome?

Measurement error in the outcome. The point of the classifier is to detect NF1 inactivation. However, the only gold standard measurement we have is NF1 mutation status. But NF1 can be inactivated in a number of different ways other than mutation! (e.g. miRNA, ALU insertion, proteasomal degradation, post-translational mRNA editing) There is no data on these samples measuring these outcomes. To me, this is what makes the machine learning task particularly interesting and I think applying the classifier as an ensemble learner actually _does_ pick up on fundamental NF1 inactivity signatures despite the performance actually appearing poor.

> For future reference, I'd consider repeated cross-validation in this situation.

I perform cv 50 times over each parameter grid. There are 15 gold standard positives and 134 gold standard negatives. We are calling these _pyrite standards_ since they are fools' gold! In other work i've extended repeated cross validation to "bootstrap holdout cross validation" and you're right, this does seem to stabilize the hyperparameter space.

> You're life will be easier if you only have to choose the regularization strength and not the mixing parameter. Consider LogisticRegressionCV (but makes you pick between lasso and ridge).

Good to know! Thanks.

> Pull request looks good to me.

@allaway  - consider the genelists frozen! :icecream:
